src/main/java/com/twilio/oai/api/JavaApiResourceBuilder.java

```diff
@@ -358,8 +358,21 @@ private ArrayList<List<CodegenParameter>> generateSignatureList(final CodegenOpe
             }
         }
         conditionalCodegenParam = Lists.cartesianProduct(conditionalCodegenParam);
-        ArrayList<List<CodegenParameter>> signatureList = new ArrayList<>();
+        List<List<CodegenParameter>> filteredConditionalCodegenParam = new ArrayList<>();
+        //Remove duplicates from signatureList
+        ArrayList<List<String>> signatureHashSet = new ArrayList<>();
         for(List<CodegenParameter> paramList : conditionalCodegenParam){
+            List<String> hash = new ArrayList<>();
```

Comment:
Can we make this List a HashSet? I am guessing from variable names, that was the original intention.

Reply:
signtureHashSet should be a HashSet since it should contain List of combination of codegenParam, and all lists need to be unique. Line 365 it should be  list and named accordingly since it would keep track the order of codegenParam. This is needed to identify different codegen param.

Reply:
> Can we make these Lists a HashSet? I am guessing from variable names, that was the original intention.

Converted this to a hashset